File: app/src/main/java/social/entourage/android/user/edit/UserEditModule.java
package social.entourage.android.user.edit;

import dagger.Module;
import dagger.Provides;

/**
 * Module related to UserEditFragment
 * Created by mihaiionescu on 01/11/16.
 */

@Module
public class UserEditModule {
    private final UserEditFragment fragment;

    public UserEditModule(final UserEditFragment fragment) {
        this.fragment = fragment;
    }

    @Provides
    public UserEditFragment providesUserEditFragment() {
        return fragment;
    }
}

File: app/src/main/java/social/entourage/android/user/edit/UserEditModule.kt
package social.entourage.android.user.edit

import dagger.Module
import dagger.Provides

/**
 * Module related to UserEditFragment
 * Created by mihaiionescu on 01/11/16.
 */
@Module
class UserEditModule(private val fragment: UserEditFragment) {
    @Provides
    fun providesUserEditFragment(): UserEditFragment {
        return fragment
    }

}